### PR TITLE
Allow StatsStore consumers to increment metrics directly

### PR DESCRIPTION
**Pull request by niik**
## Description

I found this branch laying on my machine today and took some time to clean it up while I waited for enterprise2 builds. This adds a new method called `increment` to the stats store class which lets consumers increment a numeric metric just using the name of said metric. This means we can get rid off many of our proxy methods and do away with some pretty boring boilerplate in StatsStore and Dispatcher.

I found that the `cherryPickSuccessfulWithConflictsCount` does not appear to be incremented anywhere which seemed odd but I don't believe that's my doing so I've just removed the proxy method but we should probably consider removing the metric altogether (or fixing it).

I also took the opportunity here to rewrap comments in StatStore to make it just a little bit less awful to scroll through.

## Release notes

Notes: no-notes

### CODE REVIEW

**Comment by tidy-dev** [comment]
> cherryPickSuccessfulWithConflictsCount 

My guess is this was accidentally removed during cherry pick change to multi commit operation.  I only added it because I was following the stats pattern of Rebase and figured that if we wanted data around operations that could conflict like tracked in the Merge Workflow data tiles.. We would want it for a cherry-picking as well.  If we do not care about tracking when a user successfully completes an operation with conflicts, then removing it is fine. Otherwise, I can figure out where it should be.

File: app/src/ui/dispatcher/dispatcher.ts [hunk 1]
```diff
@@ -3174,7 +2991,7 @@ export class Dispatcher {
     )
 
     if (result === CherryPickResult.CompletedWithoutError) {
-      this.statsStore.recordCherryPickSuccessfulWithConflicts()
+      this.statsStore.increment('cherryPickSuccessfulCount')
```
Comment (sergiou87) [inline app/src/ui/dispatcher/dispatcher.ts]: I think this is the one you were looking for, @niik !
```suggestion
      this.statsStore.increment('cherryPickSuccessfulWithConflictsCount')
```

File: app/src/ui/check-runs/ci-check-run-popover.tsx [hunk 2]
```diff
@@ -151,7 +151,7 @@ export class CICheckRunPopover extends React.PureComponent<
       `${this.props.repository.htmlURL}/pull/${this.props.prNumber}`
 
     this.props.dispatcher.openInBrowser(url)
-    this.props.dispatcher.recordCheckViewedOnline()
+    this.props.dispatcher.incrementMetric('opensCheckRunsPopover')
```
Comment (sergiou87) [inline app/src/ui/check-runs/ci-check-run-popover.tsx]:
```suggestion
    this.props.dispatcher.incrementMetric('viewsCheckOnline')
```

File: app/src/ui/dispatcher/dispatcher.ts [hunk 3]
```diff
@@ -142,13 +142,16 @@ export type ErrorHandler = (
...
+  public incrementMetric: StatsStore['increment']
 
   public constructor(
     private readonly appStore: AppStore,
     private readonly repositoryStateManager: RepositoryStateCache,
     private readonly statsStore: StatsStore,
     private readonly commitStatusStore: CommitStatusStore
-  ) {}
+  ) {
+    this.incrementMetric = statsStore.increment.bind(statsStore)
```
Comment (sergiou87) [inline app/src/ui/dispatcher/dispatcher.ts:153]: ❤️

File: app/src/ui/banners/rebase-conflicts-banner.tsx [hunk 4]
```diff
@@ -22,7 +22,7 @@ export class RebaseConflictsBanner extends React.Component<
   private openDialog = async () => {
     this.props.onDismissed()
     this.props.onOpenDialog()
-    this.props.dispatcher.recordRebaseConflictsDialogReopened()
+    this.props.dispatcher.incrementMetric('rebaseConflictsDialogDismissalCount')
```
Comment (sergiou87) [inline app/src/ui/banners/rebase-conflicts-banner.tsx]:
```suggestion
    this.props.dispatcher.incrementMetric('rebaseConflictsDialogReopenedCount')
```

**Review by sergiou87 — CHANGES_REQUESTED** [review]
I love this cleanup!!!

I only found a few places where a function called was replaced with an increment to the wrong property.

**Comment by niik** [comment]
> I only found a few places where a function called was replaced with an increment to the wrong property.

Thank you thank you thank you. I thought about writing some godawful tool to do this but I figured I could do it copy-pasta style without regressing. I was wrong!

**Review by sergiou87 — APPROVED** [review]
Sorry, I missed your new changes 😳  It looks all good now! :shipit:

**Pull request merged**